### PR TITLE
[YUNIKORN-2899] chore: update node version to LTS, update package and solve warnings

### PR DESCRIPTION
**Pull request by ryankert01**
### What is this PR for?
1. Browserslist: caniuse-lite is outdated. Please run:
npx update-browserslist-db@latest
Why you should do it regularly: https://github.com/browserslist/update-db#readme
2. update docusaurous to v3.5.2
3. resolve security warnings
4. update node version to latest LTS version
5. add a `pnpm run serve` for serving local built.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2899

### How should this be tested?  
tested with:
1. `./local-build.sh run`
2. `pnpm build`

need test: github ci

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Review by craigcondit — CHANGES_REQUESTED** [review]
I'm -1 on using 'lts' as a version ID. This doesn't allow us to have reproducible builds.

**Comment by ryankert01** [comment]
Hi @craigcondit, thanks for your review. 
Makes sense, I'll use 20.17 then.

**Review by craigcondit — CHANGES_REQUESTED** [review]
Please update the default version in the github action as well.

File: package.json [hunk 1]
```diff
@@ -6,18 +6,18 @@
...
-    "@docusaurus/core": "3.2.1",
-    "@docusaurus/preset-classic": "3.2.1",
-    "@docusaurus/theme-search-algolia": "3.2.1",
+    "@docusaurus/core": "3.5.2",
+    "@docusaurus/preset-classic": "3.5.2",
+    "@docusaurus/theme-search-algolia": "3.5.2",
     "@mdx-js/react": "^3.0.1",
-    "axios": "1.6.8",
-    "clsx": "^2.1.0",
-    "node": "^18.20.1",
```
Comment (craigcondit) [inline package.json:20]: Add this back in as well. This ensures that the build is reporducible.

Comment (ryankert01) [reply package.json:20]: Done.

**Comment by ryankert01** [comment]
Hi @craigcondit, just updated it and tested docker build with it.

**Comment by craigcondit** [comment]
Still missing updates to pre-commit.yml.

**Comment by ryankert01** [comment]
Gotcha!

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**